### ISSUE TITLE
handlers.mask not working when enable_validation(True)

### ISSUE THREAD

**Issue by vanAmsterdam** [pyro-ppl/numpyro#698]
Using `handlers.mask` for sampling doesn't work when `numpyro.enable_validation(True)`. 

Is this a bug or is it expected behaviour?

```python
import numpyro
numpyro.enable_validation(True) # all works fine when this is False
from jax import numpy as jnp, random
from numpyro import sample
from numpyro.handlers import mask
from numpyro.distributions import Bernoulli, Beta
from numpyro.infer.mcmc import MCMC
from numpyro.infer.hmc import NUTS

ptrue = 0.5
Y     = random.bernoulli(random.PRNGKey(0), ptrue, (100,))
Y     = jnp.where(jnp.arange(100) < 20, -1, Y)
Yobs  = Y >= 0

def m(Y, Yobs):
    theta = sample('theta', Beta(1,1))
    with numpyro.plate('obs', Y.shape[0]):
        with mask(mask_array=Yobs):
            sample('obs_Y', Bernoulli(probs=theta), obs=Y)

mcmc = MCMC(NUTS(m), 100, 500)
mcmc.run(random.PRNGKey(1), Y, Yobs)

mcmc.print_summary()
```

**Comment by fehiepsi** [comment]
I think the issue comes from: JAX has changed `0 x inf = 0` to `0 x inf = nan`.
